play-scala-rest-api-example/build.sbt

```diff
@@ -1,31 +1,45 @@
 import sbt.Keys._
 import play.sbt.PlaySettings
 
+lazy val scala213 = "2.13.10"
+lazy val scala3 = "3.3.0-RC3"
+
+lazy val core = Seq(
+  guice,
+  "org.joda" % "joda-convert" % "2.2.3",
+  "net.logstash.logback" % "logstash-logback-encoder" % "6.6",
+  "io.lemonlabs" %% "scala-uri" % "4.0.3",
+  "net.codingwell" %% "scala-guice" % "5.1.1",
+)
+
+lazy val scala2Deps = Seq(
+  "org.scalatestplus.play" %% "scalatestplus-play" % "6.0.0-M2" % Test,
+)
+
+lazy val scala3Deps= Seq(
+  "org.scalatestplus.play" %% "scalatestplus-play" % "6.0.0-M2+0-d4697b31+20230227-1631-SNAPSHOT" % Test,
+)
```

Comment:
You don't have to use two declare two different dependencies, once for Scala 2 and once for Scala 3 artifacts. A version of library gets published for different Scala version by using suffixes in the artifact's group id (either `_2.13` or `_3`).
For example: twirl-api version 1.6.0-RC2 gets published for both Scala 2.13 and Scala 3
* https://repo1.maven.org/maven2/com/typesafe/play/twirl-api_3/1.6.0-RC2/
* https://repo1.maven.org/maven2/com/typesafe/play/twirl-api_2.13/1.6.0-RC2/

sbt will automatically fetch the correct dependency based on the currently used scala version.

I will push the correct code in a couple of minutes.